src/hammer-vlsi/tech_test.py

```diff
@@ -531,6 +531,46 @@ def add_lib_with_lef(d: Dict[str, Any]) -> Dict[str, Any]:
         # Cleanup
         shutil.rmtree(tech_dir_base)
 
+    def test_special_cells(self) -> None:
+        import hammer_config
+
+        tech_dir, tech_dir_base = HammerToolTestHelpers.create_tech_dir("dummy28")
+        tech_json_filename = os.path.join(tech_dir, "dummy28.tech.json")
+
+        def add_special_cells(in_dict: Dict[str, Any]) -> Dict[str, Any]:
+            out_dict = deepdict(in_dict)
+            out_dict.update({"special_cells": [
+                                {"name": "cell1", "cell_type": "tiecell"},
+                                {"name": "cell2", "cell_type": "tiecell", "size": 1.5},
+                                {"name": "cell3", "cell_type": "iofiller", "size": 0.5},
+                                {"name": "cell4", "cell_type": "stdfiller"},
+                                {"name": "cell5", "cell_type": "endcap"},
+                             ]})
+            return out_dict
+        HammerToolTestHelpers.write_tech_json(tech_json_filename, add_special_cells)
+        tech = self.get_tech(hammer_tech.HammerTechnology.load_from_dir("dummy28", tech_dir))
+        tech.cache_dir = tech_dir
+
+        tool = DummyTool()
+        tool.technology = tech
+        database = hammer_config.HammerDatabase()
+        tool.set_database(database)
+
+        self.assertEqual(tool.technology.get_special_cell_by_type(CellType.TieCell),
+                [SpecialCell(name="cell1", cell_type=CellType.TieCell, size=None),
+                 SpecialCell(name="cell2", cell_type=CellType.TieCell, size=Decimal(1.5))
```

Comment:
`Decimal("1.5")` is preferred over `Decimal(1.5)`. In this case you get no issues because 1.5 can be exactly represented in floating point, but for other floats, you can run into issues:
```
>>> Decimal(1.5)
Decimal('1.5')
>>> Decimal(1.1)
Decimal('1.100000000000000088817841970012523233890533447265625')
```